Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

--- a/config/meson.build
+++ b/config/meson.build
@@ -6,6 +6,8 @@ elif omx_target == 'zynqultrascaleplus'
   sub = 'zynqultrascaleplus'
 elif omx_target == 'tizonia'
   sub = 'tizonia'
+ elif omx_target == 'stf'
+  sub = 'stf'
 else
   # No config file defined for the 'generic' target
   sub = ''
--- /dev/null
+++ b/config/stf/gstomx.conf
@@ -0,0 +1,17 @@
+[omxh265dec]
+type-name=GstOMXH265Dec
+core-name=/usr/lib/libsf-omx-il.so
+component-name=sf.dec.decoder.h265
+in-port-index=0
+out-port-index=1
+rank=0
+hacks=pass-profile-to-decoder;pass-color-format-to-decoder;ensure-buffer-count-actual
+
+[omxh264dec]
+type-name=GstOMXH264Dec
+core-name=/usr/lib/libsf-omx-il.so
+component-name=sf.dec.decoder.h264
+in-port-index=0
+out-port-index=1
+rank=1
+hacks=pass-profile-to-decoder;pass-color-format-to-decoder;ensure-buffer-count-actual
--- /dev/null
+++ b/config/stf/meson.build
@@ -0,0 +1 @@
+install_data (['gstomx.conf'], install_dir : omx_conf_dir)
--- a/meson.build
+++ b/meson.build
@@ -227,6 +227,8 @@ elif omx_target == 'tizonia'
   tizil_includedir = tizil_dep.get_variable('includedir')
   gst_omx_args += ['-I' + tizil_includedir + '/tizonia']
   omx_inc = []
+elif omx_target == 'stf'
+  warning('stf selected')
 else
   error ('Unsupported omx target specified. Use the -Dtarget option')
 endif
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,7 +1,7 @@
 option('header_path', type : 'string', value : '',
     description : 'An extra include directory to find the OpenMax headers')
 option('target', type : 'combo',
-    choices : ['none', 'generic', 'rpi', 'bellagio', 'tizonia', 'zynqultrascaleplus'], value : 'none',
+    choices : ['none', 'generic', 'rpi', 'bellagio', 'tizonia', 'zynqultrascaleplus', 'stf'], value : 'none',
     description : 'The OMX platform to target')
 option('struct_packing', type : 'combo',
     choices : ['0', '1', '2', '4', '8'], value : '0',
