--- a/config/stf/gstomx.conf	2021-08-18 14:51:51.384176799 +0800
+++ b/config/stf/gstomx.conf	2021-08-18 14:50:58.972186447 +0800
@@ -15,3 +15,21 @@
 out-port-index=1
 rank=1
 hacks=pass-profile-to-decoder;pass-color-format-to-decoder;ensure-buffer-count-actual
+
+[omxh265enc]
+type-name=GstOMXH265Enc
+core-name=/usr/lib/libsf-omx-il.so
+component-name=sf.enc.encoder.h265
+in-port-index=0
+out-port-index=1
+rank=1
+hacks=pass-profile-to-decoder;pass-color-format-to-decoder;ensure-buffer-count-actual;video-framerate-integer
+
+[omxh264enc]
+type-name=GstOMXH264Enc
+core-name=/usr/lib/libsf-omx-il.so
+component-name=sf.enc.encoder.h264
+in-port-index=0
+out-port-index=1
+rank=1
+hacks=pass-profile-to-decoder;pass-color-format-to-decoder;ensure-buffer-count-actual;video-framerate-integer
