--- a/config/stf/gstomx.conf
+++ b/config/stf/gstomx.conf
@@ -4,7 +4,7 @@ core-name=/usr/lib/libsf-omx-il.so
 component-name=sf.dec.decoder.h265
 in-port-index=0
 out-port-index=1
-rank=0
+rank=257
 hacks=pass-profile-to-decoder;pass-color-format-to-decoder;ensure-buffer-count-actual
 
 [omxh264dec]
@@ -13,7 +13,7 @@ core-name=/usr/lib/libsf-omx-il.so
 component-name=sf.dec.decoder.h264
 in-port-index=0
 out-port-index=1
-rank=1
+rank=257
 hacks=pass-profile-to-decoder;pass-color-format-to-decoder;ensure-buffer-count-actual
 
 [omxh265enc]
