Add NV21 for omxh264dec omxh265dec 

Signed-off-by: Leo Lu <leo.lu@starfivetech.com>

diff -purN a/omx/gstomxvideo.h b/omx/gstomxvideo.h
--- a/omx/gstomxvideo.h	2021-09-09 03:05:28.042623000 +0800
+++ b/omx/gstomxvideo.h	2022-08-23 13:45:37.042353044 +0800
@@ -36,7 +36,7 @@ G_BEGIN_DECLS
 
 /* Keep synced with gst_omx_video_get_format_from_omx(). Sort by decreasing quality */
 #define GST_OMX_VIDEO_DEC_SUPPORTED_FORMATS "{ NV16_10LE32, NV12_10LE32, " \
-  "NV16, YUY2, YVYU, UYVY, NV12, I420, RGB16, BGR16, ABGR, ARGB, GRAY8 }"
+  "NV16, YUY2, YVYU, UYVY, NV12, NV21, I420, RGB16, BGR16, ABGR, ARGB, GRAY8 }"
 
 #define GST_OMX_VIDEO_ENC_SUPPORTED_FORMATS "{ NV16_10LE32, NV12_10LE32, " \
   "NV16, NV12, I420, GRAY8 }"
