From 5ed61f2b51eea1bca9b549582f08d55ae40e7425 Mon Sep 17 00:00:00 2001
From: Stefan O'Rear <sorear@fastmail.com>
Date: Thu, 3 Sep 2020 03:17:45 -0400
Subject: [PATCH 01/16] Remove ARMSUBARCH relic from configure

---
 configure | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/configure b/configure
index 9abbbd66..acbfa8f2 100755
--- a/configure
+++ b/configure
@@ -727,11 +727,6 @@ fi
 test "$SUBARCH" \
 && printf "configured for %s variant: %s\n" "$ARCH" "$ARCH$SUBARCH"
 
-case "$ARCH$SUBARCH" in
-arm) ASMSUBARCH=el ;;
-*) ASMSUBARCH=$SUBARCH ;;
-esac
-
 #
 # Some archs (powerpc) have different possible long double formats
 # that the compiler can be configured for. The logic for whether this
-- 
2.29.2

