Upstream-Status: Inappropriate [oe-specific]
Signed-off-by: liuyingjie <yingjie.liu@spacemit.com>

diff --git a/Makefile b/Makefile
index a0fd12b9a49..cb8938cdd9e 100644
--- a/Makefile
+++ b/Makefile
@@ -1909,14 +1909,6 @@ PHONY += prepare archprepare prepare0 prepare1 prepare2 prepare3
 # and if so do:
 # 1) Check that make has not been executed in the kernel src $(srctree)
 prepare3: include/config/uboot.release
-ifneq ($(KBUILD_SRC),)
-	@$(kecho) '  Using $(srctree) as source for U-Boot'
-	$(Q)if [ -f $(srctree)/.config -o -d $(srctree)/include/config ]; then \
-		echo >&2 "  $(srctree) is not clean, please run 'make mrproper'"; \
-		echo >&2 "  in the '$(srctree)' directory.";\
-		/bin/false; \
-	fi;
-endif
 
 # prepare2 creates a makefile if using a separate output directory
 prepare2: prepare3 outputmakefile cfg
